[Live Preview in Premium/Woo themes] Add tracking events

okmttdhr opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/79223

## Proposed Changes

This PR adds tracking events for "Live Preview in Premium/Woo themes". I added the following events;

- calypso_block_theme_live_preview_upgrade_modal_open
	- This event is fired when users open the upgrade modal either by clicking the button or using the shortcut (cmd+S).
		<img width="1437" alt="Screenshot 2023-12-21 at 11 28 53" src="https://github.com/Automattic/wp-calypso/assets/5287479/9b4859f3-e78c-40b7-b684-985c6a2871ef">
		<img width="392" alt="Screenshot 2023-12-21 at 11 29 08" src="https://github.com/Automattic/wp-calypso/assets/5287479/4017d341-a55b-41bb-b912-bb10764e01c6">
- ~~calypso_block_theme_live_preview_click_upgrade~~ -> calypso_block_theme_live_preview_upgrade_modal_upgrade
	- This event is fired when users click the upgrade button on the modal. 
		<img width="1424" alt="Screenshot 2023-12-21 at 11 27 31" src="https://github.com/Automattic/wp-calypso/assets/5287479/3e9db780-0317-4343-82ed-61eb8731322f">
- calypso_block_theme_live_preview_upgrade_modal_close
	- This event is fired when the modal is closed for any reason. 


We can include the `wpcom_product_purchase` event (for checkout completion) and the `wpcom_block_editor_save_click` event (for theme activation) in funnels. 

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Please Test it on https://horizon.wordpress.com/.

* Sandbox your site and widgets.wp.com.
* Run `install-plugin.sh wpcom-block-editor add/live-preview-track-event` on your sandbox.
* Go to the Theme Showcase page and pick a Premium or Woo theme.
* Click the `Upgrade now` button.
* Verify the event is fired.
	* When you click the button on the header in the edit mode.
		<img width="746" alt="Screenshot 2023-12-21 at 11 19 50" src="https://github.com/Automattic/wp-calypso/assets/5287479/ff1593ec-e2d7-4e9f-ac1c-6961cfa43ab6">
	* When you click the button on the sidebar.
		<img width="746" alt="Screenshot 2023-12-21 at 11 19 34" src="https://github.com/Automattic/wp-calypso/assets/5287479/c1961744-287b-45d6-9f4b-60847952fcb0">
	* When you use the shortcut.
		<img width="750" alt="Screenshot 2023-12-21 at 11 32 13" src="https://github.com/Automattic/wp-calypso/assets/5287479/e37b4cc7-34ce-4d84-8dd7-6213cb7c8765">
* Verify the event is fired when you close the modal by
	* Clicking the close icon
		<img width="750" alt="Screenshot 2023-12-22 at 11 13 56" src="https://github.com/Automattic/wp-calypso/assets/5287479/4af880ca-444a-4806-9eaa-e76f346cdaaa">
	* Clicking the Cancel button
		<img width="745" alt="Screenshot 2023-12-22 at 11 14 06" src="https://github.com/Automattic/wp-calypso/assets/5287479/0825fec3-2799-4431-a5d1-e0bc52bddca1">
	* Clicking the overlay or typing the `escape` key
		<img width="747" alt="Screenshot 2023-12-22 at 11 14 15" src="https://github.com/Automattic/wp-calypso/assets/5287479/46276097-3220-427a-b067-51a368e60dd6">
* Click the `Upgrade plan` button on the modal.
* Verify the event is fired.
	<img width="751" alt="Screenshot 2023-12-22 at 11 15 11" src="https://github.com/Automattic/wp-calypso/assets/5287479/ed279b6c-970e-4a26-816e-493303706404">

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

okmttdhr committed:
Add upgrade_modal_open event

okmttdhr committed:
Add click_upgrade event

okmttdhr committed:
Add the canvasMode prop

taipeicoder commented:
Using the shortcut returns `canvas_mode: view` while in edit mode:
![Screenshot 2023-12-21 at 11 17 34 AM](https://github.com/Automattic/wp-calypso/assets/797888/586bfdd5-16ef-499a-9b3a-2c3e8142f1f0)

taipeicoder commented:
We could also track when upgrade modal is closed via user-interaction:
- Clicking the Cancel button.
- Clicking the Close button.
- Clicking the backdrop or pressing the <kbd>Esc</kbd> button.

WDYT? @okmttdhr @autumnfjeld

autumnfjeld commented:
> calypso_block_theme_live_preview_click_upgrade
This event is fired when users click the upgrade button on the modal.

This event should include the word modal, maybe:  `calypso_block_theme_live_preview_upgrade_modal_click_upgrade`

You can see other modal events by searching for "modal" in tracks events.

autumnfjeld commented:
> We could also track when upgrade modal is closed via user-interaction:
> 
> * Clicking the Cancel button.
> * Clicking the Close button.
> * Clicking the backdrop or pressing the Esc button.
> 
> WDYT? @okmttdhr @autumnfjeld

That could be interesting, like a measure of curiosity about upgrading.

arthur791004 reviewed (approved):
The code looks good 👍

okmttdhr committed:
Add the modal closed event

okmttdhr committed:
Fix the canvasMode is not properly tracked

okmttdhr commented:
@taipeicoder @autumnfjeld 
> Using the shortcut returns `canvas_mode: view` while in edit mode: ![Screenshot 2023-12-21 at 11 17 34 AM](https://private-user-images.githubusercontent.com/797888/292077260-586bfdd5-16ef-499a-9b3a-2c3e8142f1f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMyMDcyMTIsIm5iZiI6MTcwMzIwNjkxMiwicGF0aCI6Ii83OTc4ODgvMjkyMDc3MjYwLTU4NmJmZGQ1LTE2ZWYtNDk5YS05YjNhLTJjM2U4MTQyZjFmMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyMlQwMTAxNTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01Y2MzYjYzZDY4ZjcxODgwMDc1NThiNGJkMDEyZjA0NjE1ZWM5ZTE0OGZlYzBjMzZiZWJhZWJjZTBkMjJhZDU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.9VR0nqCDqmp13_QYe-KrD1GodVjxBHQAXgKEYtXLBW8)

I addressed this. Thank you for pointing this out!

>We could also track when upgrade modal is closed via user-interaction:
>This event should include the word modal, maybe: calypso_block_theme_live_preview_upgrade_modal_upgradeclick

I added `calypso_block_theme_live_preview_upgrade_modal_close` with `closed_by='close_icon' | 'cancel_button' | 'dialog_action'`, and renamed `calypso_block_theme_live_preview_click_upgrade` to `calypso_block_theme_live_preview_upgrade_modal_upgrade`, aligning with `calypso_block_theme_live_preview_upgrade_modal_close`. 🙂

taipeicoder reviewed (approved):
LGTM 🙂
